models/quoteCollection.js

```diff
@@ -47,7 +49,8 @@ Object.assign(quoteCollection.methods, {
 
   getQuotes() {
     const Quote = mongoose.model('Quote');
-    return Promise.all(this.quotes.map(id => Quote.findById(id)));
+    return Promise.all(this.quotes.map(id => Quote.findById(id)))
+      .then(quotes => quotes.filter((quote) => quote !== null));
```

Comment:
This is suboptimal from database perspective. You are looping over array of ids asking mongo each time to retrieve single document. Instead you should use `Quote.find` with array of possible ids (issues single mongo query). :warning: 
Also, this filter looks weird :dizzy_face: